Address remaining Safer CPP failures in WebViewImpl.mm

cdumez opened the pull request:
#### d5c0d74713aa1861f2cc0451d0bc5f9b7f7e5f6b
<pre>
Address remaining Safer CPP failures in WebViewImpl.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=289348">https://bugs.webkit.org/show_bug.cgi?id=289348</a>

Reviewed by Geoffrey Garen.

* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::supportsType):
(WebCore::Image::protectedData const):
* Source/WebCore/platform/graphics/Image.h:
* Source/WebKit/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(isType):
(-[WKTextListTouchBarViewController setCurrentListType:]):
(-[WKTextTouchBarItemController itemForIdentifier:]):
(WebKit::m_flagsChangedEventMonitorTrackingArea):
(WebKit::WebViewImpl::~WebViewImpl):
(WebKit::WebViewImpl::processWillSwap):
(WebKit::WebViewImpl::didRelaunchProcess):
(WebKit::WebViewImpl::screenDidChangeColorSpace):
(WebKit::WebViewImpl::setInspectorAttachmentView):
(WebKit::WebViewImpl::remoteObjectRegistry):
(WebKit::WebViewImpl::startDrag):
(WebKit::WebViewImpl::provideDataForPasteboard):
(WebKit::WebViewImpl::namesOfPromisedFilesDroppedAtDestination):
(WebKit::WebViewImpl::takeViewSnapshot):
(WebKit::WebViewImpl::saveBackForwardSnapshotForCurrentItem):
(WebKit::WebViewImpl::ensureProtectedGestureController):
(WebKit::WebViewImpl::magnification const):
(WebKit::WebViewImpl::setCustomSwipeViews):
(WebKit::WebViewImpl::setCustomSwipeViewsTopContentInset):
(WebKit::WebViewImpl::tryToSwipeWithEvent):
(WebKit::WebViewImpl::setDidMoveSwipeSnapshotCallback):
(WebKit::WebViewImpl::scrollWheel):
(WebKit::WebViewImpl::smartMagnifyWithEvent):
(WebKit::WebViewImpl::didRestoreScrollPosition):
(WebKit::WebViewImpl::beginBackSwipeForTesting):
(WebKit::WebViewImpl::completeBackSwipeForTesting):
(WebKit::WebViewImpl::dismissTextTouchBarPopoverItemWithIdentifier):
(WebKit::WebViewImpl::updateMediaPlaybackControlsManager):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _dismissTextTouchBarPopoverItemWithIdentifier:]):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/291825@main">https://commits.webkit.org/291825@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a481933d5bcde2736d0e6e961544fdecb969aef8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94102 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13689 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3435 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99113 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44629 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96152 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13989 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22119 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71791 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29136 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97104 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10374 "Found 1 new test failure: fast/forms/ios/focus-input-via-button-no-scaling.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84971 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52132 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10059 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2650 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43978 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80303 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2738 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101152 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21154 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15401 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80796 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21406 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80976 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80172 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24718 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2082 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/14316 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15095 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21138 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26317 "Found 0 new failure in UIProcess/mac/WebViewImpl.mm") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20825 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24285 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22566 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebCore/platform/mac/AppKitExtras.h:
```diff
@@ -0,0 +1,53 @@
+
+SPECIALIZE_TYPE_TRAITS_BEGIN(NSPopoverTouchBarItem)
+static bool isType(const NSObject& object) { return [&object isKindOfClass:[NSPopoverTouchBarItem class]]; }
+SPECIALIZE_TYPE_TRAITS_END()
```
This is fine, but isn't it more idiomatic to use `dynamic_objc_cast`, `checked_objc_cast`, and `is_objc`?

rniwa replied:
Yeah, we should be using `dynamic_objc_cast` & `is_objc` instead.

cdumez replied:
Oh, I wasn't familiar with this! I'll update. Thanks.

cdumez committed:
Address remaining Safer CPP failures in WebViewImpl.mm
https://bugs.webkit.org/show_bug.cgi?id=289348

Reviewed by Geoffrey Garen.

* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::supportsType):
(WebCore::Image::protectedData const):
* Source/WebCore/platform/graphics/Image.h:
* Source/WebKit/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(isType):
(-[WKTextListTouchBarViewController setCurrentListType:]):
(-[WKTextTouchBarItemController itemForIdentifier:]):
(WebKit::m_flagsChangedEventMonitorTrackingArea):
(WebKit::WebViewImpl::~WebViewImpl):
(WebKit::WebViewImpl::processWillSwap):
(WebKit::WebViewImpl::didRelaunchProcess):
(WebKit::WebViewImpl::screenDidChangeColorSpace):
(WebKit::WebViewImpl::setInspectorAttachmentView):
(WebKit::WebViewImpl::remoteObjectRegistry):
(WebKit::WebViewImpl::startDrag):
(WebKit::WebViewImpl::provideDataForPasteboard):
(WebKit::WebViewImpl::namesOfPromisedFilesDroppedAtDestination):
(WebKit::WebViewImpl::takeViewSnapshot):
(WebKit::WebViewImpl::saveBackForwardSnapshotForCurrentItem):
(WebKit::WebViewImpl::ensureProtectedGestureController):
(WebKit::WebViewImpl::magnification const):
(WebKit::WebViewImpl::setCustomSwipeViews):
(WebKit::WebViewImpl::setCustomSwipeViewsTopContentInset):
(WebKit::WebViewImpl::tryToSwipeWithEvent):
(WebKit::WebViewImpl::setDidMoveSwipeSnapshotCallback):
(WebKit::WebViewImpl::scrollWheel):
(WebKit::WebViewImpl::smartMagnifyWithEvent):
(WebKit::WebViewImpl::didRestoreScrollPosition):
(WebKit::WebViewImpl::beginBackSwipeForTesting):
(WebKit::WebViewImpl::completeBackSwipeForTesting):
(WebKit::WebViewImpl::dismissTextTouchBarPopoverItemWithIdentifier):
(WebKit::WebViewImpl::updateMediaPlaybackControlsManager):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _dismissTextTouchBarPopoverItemWithIdentifier:]):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/291825@main
